src/thriveopedia/pages/ThriveopediaMuseumPage.cs

```diff
@@ -0,0 +1,178 @@
+                }
+
+                card.SavedSpecies = (MicrobeSpecies)FossilisedSpecies.LoadSpeciesFromFile(speciesName);
+                card.Connect(nameof(MuseumCard.OnSpeciesSelected), this, nameof(UpdateSpeciesPreview));
+                cardContainer.AddChild(card);
+            }
+        }
+    }
+
+    public override void UpdateCurrentWorldDetails()
+    {
+    }
+
+    public override void OnNavigationPanelSizeChanged(bool collapsed)
+    {
+        // Change the number of columns to reflect the space the museum takes up
+        cardContainer.Columns = collapsed ? 3 : 4;
+    }
+
+    private void UpdateSpeciesPreview(MuseumCard card)
+    {
+        if (!speciesPreviewContainer.Visible)
+        {
+            welcomeLabel.Visible = false;
+            speciesPreviewContainer.Visible = true;
+        }
+
+        var species = card.SavedSpecies;
+        speciesPreview.PreviewSpecies = species;
+
+        // Deselect all other cards to prevent highlights hanging around.
+        foreach (MuseumCard otherCard in cardContainer.GetChildren())
+        {
+            if (otherCard != card)
+                otherCard.Pressed = false;
+        }
+
+        if (species is MicrobeSpecies microbeSpecies)
+        {
+            hexesPreview.PreviewSpecies = microbeSpecies;
+        }
+        else
+        {
+            GD.PrintErr("Unknown species type to preview: ", species);
+        }
+
+        UpdateSpeciesDetail(species);
+    }
+
+    private void UpdateSpeciesDetail(Species species)
+    {
+        speciesDetailsLabel.ExtendedBbcode = TranslationServer.Translate("SPECIES_DETAIL_TEXT").FormatSafe(
+            species.FormattedNameBbCode, species.ID, species.Generation, species.Population, species.Colour.ToHtml(),
+            string.Join("\n  ", species.Behaviour.Select(b =>
+                BehaviourDictionary.GetBehaviourLocalizedString(b.Key) + ": " + b.Value)));
+
+        switch (species)
+        {
+            case MicrobeSpecies microbeSpecies:
+            {
+                speciesDetailsLabel.ExtendedBbcode += "\n" +
+                    TranslationServer.Translate("MICROBE_SPECIES_DETAIL_TEXT").FormatSafe(
+                        microbeSpecies.MembraneType.Name, microbeSpecies.MembraneRigidity,
+                        microbeSpecies.BaseSpeed, microbeSpecies.BaseRotationSpeed, microbeSpecies.BaseHexSize);
+                break;
+            }
+        }
+    }
+
+    private void OnOpenInFreebuildPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        if (speciesPreview.PreviewSpecies == null)
+            return;
+
+        // If we're opening from a game in progress, warn the player
+        if (CurrentGame != null)
+        {
+            leaveGameConfirmationDialog.DialogText = TranslationServer.Translate("OPEN_IN_FREEBUILD_WARNING");
+            leaveGameConfirmationDialog.PopupCenteredShrink();
+            return;
+        }
+
+        TransitionToFreebuild();
```

Comment:
I'd say add the previewSpecies as parameter here, then the nullability suppression in that method is not necessary for cleaner code.